### ISSUE TITLE
Bug: id-denylist throw an error on property

### ISSUE THREAD

**Issue by jon9090** [eslint/eslint#16221]
### Environment

Node version: v16.15.1
npm version: 8.11.0
Local ESLint version: ~8.15.0
Global ESLint version: 
Operating System: Mac Pro

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```
"id-denylist": ["error", "data", "err", "e", "cb", "callback"]
```
</details>

```typescript
 const [error, response] = await awaitTo<AxiosResponse<any>, AxiosError<any>>(
    axios({
      url: `/api/send`,
      method: 'post',
      data: {},
      httpsAgent: new Agent({ rejectUnauthorized: false }),
      auth: { username: user.id, password: user.password },
    })
  );

```

### What did you expect to happen?

Axios have `data` property. eslint say it's restricted. I was expecting [`id-denylist`](https://eslint.org/docs/latest/rules/id-denylist) to throw the exception only on declaration. 
Obviously I can't change axios interface. so it prevent me to use this plugin right now.

### What actually happened?

<img width="582" alt="image" src="https://user-images.githubusercontent.com/74149517/185732632-6ed7920c-169a-4007-85e0-e30728cfc88b.png">

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by JounQin** [comment]
> This rule will catch disallowed identifiers that are:
> * ...
> * object properties assigned to during object creation
> * ...

**Comment by snitin315** [comment]
This [works as intended](https://eslint.org/docs/latest/rules/id-denylist#rule-details). Since this is 3rd party API it makes sense to disable it for this particular instance.

```js
 const [error, response] = await awaitTo<AxiosResponse<any>, AxiosError<any>>(
    axios({
      url: `/api/send`,
      method: 'post',
      //eslint-disable-next-line id-denylist
      data: {},
      httpsAgent: new Agent({ rejectUnauthorized: false }),
      auth: { username: user.id, password: user.password },
    })
  );
```
